subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/plugins/PluginBuildsIntegrationTest.groovy

```diff
@@ -624,4 +625,114 @@ class PluginBuildsIntegrationTest extends AbstractPluginBuildIntegrationTest {
         where:
         dsl << ['Groovy', 'Kotlin']
     }
+
+    def "multiple included plugin builds which each include a library build with the same name works with #dsl dsl"() {
+        given: "a plugin with a dependency on a library build named 'lib'"
+        file("plugin1/lib/build.gradle") << """
+            plugins {
+                id("java-library")
+            }
+        """
+
+        def pluginBuild1 = pluginBuild("plugin1", dsl == 'Kotlin')
+        pluginBuild1.settingsFile.text = """
+            include(":lib")
+        """
+        pluginBuild1.buildFile << """
+            dependencies {
+                implementation(project(":lib"))
+            }
+        """
+
+        and: "another plugin with a dependency on another library build with the same name"
+        file("plugin2/lib/build.gradle") << """
+            plugins {
+                id("java-library")
+            }
+        """
+
+        def pluginBuild2 = pluginBuild("plugin2", dsl == 'Kotlin')
+        pluginBuild2.settingsFile.text = """
+            include(":lib")
+        """
+        pluginBuild2.buildFile << """
+            dependencies {
+                implementation(project(":lib"))
+            }
+        """
+
+        and: "both plugins are included in the root build"
+        settingsFile.text = """
+            pluginManagement {
+                includeBuild("${pluginBuild1.buildName}")
+                includeBuild("${pluginBuild2.buildName}")
+            }
+        """
+        buildFile << """
+            plugins {
+                id("${pluginBuild1.projectPluginId}")
+                id("${pluginBuild2.projectPluginId}")
+            }
+        """
+
+        expect:
+        succeeds()
+
+        where:
+        dsl << ['Groovy', 'Kotlin']
+    }
+
+    def "multiple included plugin builds which each include a plugin build with the same name fails with #dsl dsl"() {
+        given: "a plugin that applies an included plugin build named 'lib'"
+        def libBuild1 = pluginBuild("plugin1", "lib", dsl == 'Kotlin')
+
+        def pluginBuild1 = pluginBuild("plugin1", dsl == 'Kotlin')
+        pluginBuild1.settingsFile.text = """
+            pluginManagement {
+                includeBuild("${libBuild1.buildName}")
+            }
+        """
+        pluginBuild1.buildFile << """
+            plugins {
+                id("${libBuild1.projectPluginId}")
+            }
+        """
```

Comment:
isn't this what the test fixture should already be doing?

oh, I think I see... libBuild1 has a plugin, but it's nested deeper.  pluginBuild1 has a plugin.

You're adding libBuild1 as another included build, which should have the same plugin id. I think this is a case where the test fixtures are doing us no favors.